CIP-0062/README.md

```diff
@@ -0,0 +1,473 @@
+
+This also addresses some short-comings of [CIP-30](https://cips.cardano.org/cips/cip30/); which signData can only be done by known an address; This signature is not relevant to a specific address, nor the dApp will know an address attached to the voting key. The voting key derivation is defined in [CIP-36](https://cips.cardano.org/cips/cip36/).
+
+Perhaps [CIP-30](https://cips.cardano.org/cips/cip30/) could be expanded to also know how to perform `signData` from a given public key from a specific derivation path; instead of doing so only by known address.
+
+The other reason for this specification is to have a specific, but optional, namespace for governance specific wallet functionality. As such, wallet providers might choose not to implement this specification on top of [CIP-30](https://cips.cardano.org/cips/cip30/).
+
+# Specification
+
+## Version
+
+The API Extension specified in this document will count as version 0.2.0 for version-checking purposes below.
+
+## Data Types
+
+### PublicKey
+
+A hex string representing a 32 byte Ed25519 public key.
+
+### GovernanceKey
+
+```ts
+type GovernanceKey = {
+  votingKey: PublicKey
+  weight: number
+}
+```
+
+* `votingKey` - A voting `PublicKey` used to represent the target of the delegation.
+* `weight` - Used to calculate the actual voting power using the rules described
+  in [CIP-36](https://cips.cardano.org/cips/cip36/).
+
+### Voting Purpose
+
+```ts
+type enum VotingPurpose = {
+  CATALYST = 0
+}
+```
+
+`VotingPurpose`: Defines the voting purpose of the governance functions. This is used
+to limit the scope of the governance functions. For example, a voting purpose
+might be a subset of Catalyst proposals, a council election, or even some
+private purpose (agreed by convention). Currently, only voting purpose 0 (Zero)
+is defined, and it is for Catalyst events.
+
+**IMPORTANT**: The List of purposes in this CIP should be considered the
+authoritative list of known purposes, subject to future amendment. Other voting
+purposes will be defined as required, by either an update to this CIP or a
+future CIP listing currently allocated Voting Purposes.
+
+### VotingCredentials
+```ts
+interface VotingCredentials {
+  votingKey: PublicKey
+  stakingCredential: PublicKey
+}
+```
+Information used to represent a wallet's voting credentials.
+
+* votingKey - Derivation as described within [CIP-0036](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0036), wallets should use `address_index`= 0.
+* stakingCredential - At the moment, the only supported staking credential is a public staking key. This is used to represent the wallet's staked ADA.
+
+### BlockDate
+
+```ts
+interface BlockDate {
+  epoch: number
+  slot: number
+}
+```
+
+* epoch - an epoch value on the voting block chain.
+* slot - a block slot number on the voting block chain.
+
+#### Proposal
+
+```ts
+interface Proposal {
+  votePlanId: string;
+  proposalIndex: number;
+  voteOptions?: number[];
+  voteEncKey?: string;
+}
```

Comment:
since voteEncKey and voteOptions are required for private votes, I'd consider rewriting it like this to make that requirement explicit at type level:

```suggestion
interface Proposal =
  | {
      votePlanId: HexString
      voteOptions: number[]
      votePublic: false
      proposalIndex: number
      voteEncKey: HexString
    }
  | {
      votePlanId: HexString
      voteOptions: number[]
      votePublic: true
      proposalIndex: number
    }
```

Reply:
This is sensible - I will make this change.